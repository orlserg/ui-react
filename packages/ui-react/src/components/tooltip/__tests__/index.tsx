import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react';

import { Tooltip } from '..';

test('Default use', async () => {
    const { asFragment, getByText } = render(
        <Tooltip
            children="Text trigger"
            render="Text render"
        />,
        {
            container: document.body
        }
    );

    expect(asFragment()).toMatchSnapshot();

    fireEvent.click(getByText('Text trigger'));

    await waitFor(() => {
        expect(asFragment()).toMatchSnapshot();
    });
});

test('Close', async () => {
    const { asFragment, getByText } = render(
        <div>
            <Tooltip
                children="Text trigger"
                render="Text render"
            />
            <div>Out click</div>
        </div>,
        {
            container: document.body
        }
    );

    expect(asFragment()).toMatchSnapshot();

    fireEvent.click(getByText('Text trigger'));

    await waitFor(() => {
        expect(asFragment()).toMatchSnapshot();
    });

    fireEvent.click(getByText('Out click'));

    await waitFor(() => {
        expect(asFragment()).toMatchSnapshot();
    });
});

test('Element', async () => {
    const { asFragment, getByText } = render(
        <Tooltip
            children={<a href="#">Text trigger</a>}
            render="Text render"
        />,
        {
            container: document.body
        }
    );

    expect(asFragment()).toMatchSnapshot();

    fireEvent.click(getByText('Text trigger'));

    await waitFor(() => {
        expect(asFragment()).toMatchSnapshot();
    });
});

test('Hover', async () => {
    const { asFragment, getByText } = render(
        <Tooltip
            children="Text trigger"
            render="Text render"
            trigger="hover"
        />
    );

    expect(asFragment()).toMatchSnapshot();

    fireEvent.click(getByText('Text trigger'));

    await waitFor(() => {
        expect(asFragment()).toMatchSnapshot();
    });
});

test('Render function', async () => {
    const { asFragment, getByText } = render(
        <Tooltip
            children="Text trigger"
            render={() => 'Text render'}
        />,
        {
            container: document.body
        }
    );

    expect(asFragment()).toMatchSnapshot();

    fireEvent.click(getByText('Text trigger'));

    await waitFor(() => {
        expect(asFragment()).toMatchSnapshot();
    });
});
