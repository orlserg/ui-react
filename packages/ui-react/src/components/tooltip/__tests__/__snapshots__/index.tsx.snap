// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Close 1`] = `
<DocumentFragment>
  <div>
    <div
      class="popover__trigger"
    >
      Text trigger
    </div>
    <div>
      Out click
    </div>
  </div>
</DocumentFragment>
`;

exports[`Close 2`] = `
<DocumentFragment>
  <div>
    <div
      class="popover__trigger"
    >
      Text trigger
    </div>
    <div>
      Out click
    </div>
  </div>
</DocumentFragment>
`;

exports[`Close 3`] = `
<DocumentFragment>
  <div>
    <div
      class="popover__trigger"
    >
      Text trigger
    </div>
    <div>
      Out click
    </div>
  </div>
</DocumentFragment>
`;

exports[`Default use 1`] = `
<DocumentFragment>
  <div
    class="popover__trigger"
  >
    Text trigger
  </div>
</DocumentFragment>
`;

exports[`Default use 2`] = `
<DocumentFragment>
  <div
    class="popover__trigger"
  >
    Text trigger
  </div>
</DocumentFragment>
`;

exports[`Element 1`] = `
<DocumentFragment>
  <span
    class="popover__trigger"
  >
    <a
      href="#"
    >
      Text trigger
    </a>
  </span>
</DocumentFragment>
`;

exports[`Element 2`] = `
<DocumentFragment>
  <span
    class="popover__trigger"
  >
    <a
      href="#"
    >
      Text trigger
    </a>
  </span>
</DocumentFragment>
`;

exports[`Hover 1`] = `
<DocumentFragment>
  <div
    class="popover__trigger"
  >
    Text trigger
  </div>
</DocumentFragment>
`;

exports[`Hover 2`] = `
<DocumentFragment>
  <div
    class="popover__trigger"
  >
    Text trigger
  </div>
</DocumentFragment>
`;

exports[`Render function 1`] = `
<DocumentFragment>
  <div
    class="popover__trigger"
  >
    Text trigger
  </div>
</DocumentFragment>
`;

exports[`Render function 2`] = `
<DocumentFragment>
  <div
    class="popover__trigger"
  >
    Text trigger
  </div>
</DocumentFragment>
`;
